Implement Controller

german-e-mas opened the pull request:
In this PR I implement the controller side of the MVC Pattern.

Updates include:
- Controller class.
- Speed of the Aircraft was changed.
- Radius updates.
- Minor language correction.
- Touch Event to select Aircraft and move them around.

ptal @pablisho

pablisho commented on app/src/main/java/mas/german/landingplanes/Game.java:
```diff
@@ -24,7 +24,9 @@
-    private static final int UPDATE_MS = 200;
+    private static final int UPDATE_MS = 30;
+    // Modifier of the Aircraft radius, in order to give a larger margin of selection.
+    private static final float SIZE_MULTIPLIER = 1.25f;
```
I wouldn't think it this way. I would think it as a tolerance or distance from the centre, similar as the landing.

german-e-mas replied:
I agree, it is more of a tolerance than a size modifier. `DISTANCE_TOLERANCE` seems like a better name for what it does.

pablisho commented on app/src/main/java/mas/german/landingplanes/Game.java:
```diff
@@ -86,6 +88,19 @@
+        /**
+         * A specific aircraft has been deselected.
+         */
+        void onAircraftDeselect(int id);
```
Maybe better to have only one callback and pass a boolean "selected" as param indicating if it was selected or deselected.

pablisho commented on app/src/main/java/mas/german/landingplanes/Game.java:
```diff
@@ -261,6 +286,65 @@ public Aerodrome getAerodrome() {
+        synchronized (mAircraftList) {
+            for (Aircraft aircraft : mAircraftList) {
+                if (aircraft.getId() == id) {
+                    aircraft.select();
```
same here, maybe aircraft.select(true) aircraft.select(false) could be better

pablisho commented on app/src/main/java/mas/german/landingplanes/MainActivity.java:
```diff
@@ -2,15 +2,21 @@
     private static final String TAG = MainActivity.class.getSimpleName();
+    private AerodromeView mView;
     private Game mGame = Game.getInstance();
+    private Controller mCtrl = new Controller();
```
mController

pablisho commented on app/src/main/java/mas/german/landingplanes/MainActivity.java:
```diff
@@ -2,15 +2,21 @@
         super.onCreate(savedInstanceState);
         setContentView(R.layout.activity_main);
+        mView = (AerodromeView) findViewById(R.id.view_aerodrome);
+        mView.setListener(mCtrl);
```
setController

pablisho commented on app/src/main/java/mas/german/landingplanes/view/AerodromeView.java:
```diff
@@ -131,6 +167,30 @@ public void onAircraftPositionChanged() {
+        if (aircraftDrawable.getId() == id) {
+          aircraftDrawable.select();
+        } else {
+          aircraftDrawable.deselect();
```
same about select/deselect

german-e-mas commented:
I addressed the comments @pablisho . I unified the selection and de-selection methods of both `Aircraft` and `AircraftDrawable`, along various comments and renames.

ptal again.

german-e-mas commented:
I changed the distance tolerance to an addition, not a multiplication.

ptal @pablisho
